src/core/vector/qgsvectorlayerjoinbuffer.cpp

```diff
@@ -172,9 +172,27 @@ void QgsVectorLayerJoinBuffer::cacheJoinLayer( QgsVectorLayerJoinInfo &joinInfo
       }
       else
       {
-        QgsAttributes attrs2 = attrs;
-        attrs2.remove( joinFieldIndex );  // skip the join field to avoid double field names (fields often have the same name)
-        joinInfo.cachedAttributes.insert( key, attrs2 );
+        QgsAttributes attributesCache;
+        for ( int i = 0; i < attrs.size(); i++ )
+        {
+          if ( i == joinFieldIndex )
+            continue;
+
+          if ( !joinInfo.prefix().isNull() ) // Default prefix 'layerName_' used
```

Comment:
isn't there a possible clash even when using layerName_fieldname? I think it'd be safest to also check for clashes even when there's no prefix

Reply:
Yes it was theoretically possible, I have changed it know.